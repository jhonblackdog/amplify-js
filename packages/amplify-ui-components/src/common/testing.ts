export const pixelThreshold = 10;
