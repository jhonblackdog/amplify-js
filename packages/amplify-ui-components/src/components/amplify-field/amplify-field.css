amplify-field input {
  padding: 5px;
  width: 100%;
  max-width: 300px;
  box-sizing: border-box;
  border: 1px gray solid;
}

amplify-field input:focus {
  outline: none;
  border-color: orange;
}

amplify-field .label {
  display: block;
  font-size: 0.9em;
  font-weight: bold;
  padding-bottom: 0.5em;
}

amplify-field .description {
  font-size: 0.8em;
  padding-top: 0.5em;
}
