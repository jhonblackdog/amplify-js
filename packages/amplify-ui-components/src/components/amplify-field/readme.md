# amplify-field



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description                                                           | Type                    | Default     |
| ------------- | ------------- | --------------------------------------------------------------------- | ----------------------- | ----------- |
| `description` | `description` | The text of the description.  Goes just below the label.              | `string`                | `undefined` |
| `fieldId`     | `field-id`    | The ID of the field.  Should match with its corresponding input's ID. | `string`                | `undefined` |
| `label`       | `label`       | The text of the label.  Goes above the input. Ex: "First name"        | `string`                | `undefined` |
| `onInput`     | --            | The callback, called when the input is modified by the user.          | `(arg0: Event) => void` | `undefined` |
| `type`        | `type`        | The input type.  Can be any HTML input type.                          | `string`                | `"text"`    |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)

### Depends on

- [amplify-input](../amplify-input)

### Graph
```mermaid
graph TD;
  amplify-field --> amplify-input
  amplify-examples --> amplify-field
  style amplify-field fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
