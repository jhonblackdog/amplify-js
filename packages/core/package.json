{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
