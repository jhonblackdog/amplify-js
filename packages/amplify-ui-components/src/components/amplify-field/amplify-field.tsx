import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'amplify-field',
  styleUrl: 'amplify-field.css',
})
export class AmplifyField {
  @Prop() fieldId: string;
  @Prop() label: string | null;
  @Prop() description: string | null;
  @Prop() type?: string = "text";
  @Prop() onInput?: (arg0: Event) => void;

  render() {
    return (
      <div>
        {this.label && (
          <label htmlFor={this.fieldId} class="label">
            {this.label}
          </label>
        )}
        {this.description && (
          <div id={`${this.fieldId}-description`} class="description">
            {this.description}
          </div>
        )}
        <div>
          <amplify-input
            fieldId={this.fieldId}
            aria-describedby={this.fieldId && this.description ? `${this.fieldId}-description` : null}
            type={this.type}
            onInput={this.onInput}
          />
        </div>
      </div>
    );
  }
}
