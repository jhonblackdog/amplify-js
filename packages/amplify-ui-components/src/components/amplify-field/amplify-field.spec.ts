import { newSpecPage } from '@stencil/core/testing';
import { AmplifyField } from './amplify-field.tsx'

describe('amplify-field', () => {
  it('renders no label or description if none are provided', async () => {
    const page = await newSpecPage({
      components: [AmplifyField],
      html: `<amplify-field></amplify-field>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders a label and description, if they are provided, and no id', async () => {
    const page = await newSpecPage({
      components: [AmplifyField],
      html: `<amplify-field label='label' description='description'></amplify-field>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders with an id, if it is provided', async () => {
    const page = await newSpecPage({
      components: [AmplifyField],
      html: `<amplify-field label='label' description='description' field-id='id'></amplify-field>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
