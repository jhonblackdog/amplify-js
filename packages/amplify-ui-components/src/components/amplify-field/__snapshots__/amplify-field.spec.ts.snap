// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-field renders a label and description, if they are provided, and no id 1`] = `
<amplify-field description="description" label="label">
  <!---->
  <div>
    <label class="label">
      label
    </label>
    <div class="description" id="undefined-description">
      description
    </div>
    <div>
      <slot-fb name="input">
        <amplify-input type="text"></amplify-input>
      </slot-fb>
    </div>
  </div>
</amplify-field>
`;

exports[`amplify-field renders no label or description if none are provided 1`] = `
<amplify-field>
  <!---->
  <div>
    <div>
      <slot-fb name="input">
        <amplify-input type="text"></amplify-input>
      </slot-fb>
    </div>
  </div>
</amplify-field>
`;

exports[`amplify-field renders with an id, if it is provided 1`] = `
<amplify-field description="description" field-id="id" label="label">
  <!---->
  <div>
    <label class="label" htmlfor="id">
      label
    </label>
    <div class="description" id="id-description">
      description
    </div>
    <div>
      <slot-fb name="input">
        <amplify-input aria-describedby="id-description" fieldid="id" type="text"></amplify-input>
      </slot-fb>
    </div>
  </div>
</amplify-field>
`;

exports[`amplify-field replaces the input component, if a new one is provided 1`] = `
<amplify-field description="description" field-id="id" label="label">
  <!---->
  <div>
    <label class="label" htmlfor="id">
      label
    </label>
    <div class="description" id="id-description">
      description
    </div>
    <div>
      <input slot="input">
      <slot-fb hidden="" name="input">
        <amplify-input aria-describedby="id-description" fieldid="id" type="text"></amplify-input>
      </slot-fb>
    </div>
  </div>
</amplify-field>
`;
